sunpy/net/dataretriever/sources/aia_synoptic.py

```diff
@@ -0,0 +1,195 @@
+import numpy as np
+import re
+from datetime import datetime, timedelta
+from sunpy.net.dataretriever import GenericClient
+from sunpy.net import attrs as a
+import astropy.units as u
+from sunpy.net.dataretriever.client import QueryResponse
+
+__all__ = ["AIASynopticClient"]
+
+
+class AIASynopticClient(GenericClient):
+    """
+    A client for retrieving AIA synoptic data from JSOC.
+
+    This client retrieves synoptic AIA data from the following source:
+    https://jsoc1.stanford.edu/data/aia/synoptic/
+
+    The synoptic dataset includes lower-resolution (1k) images, with additional
+    image processing steps like downsampling and time integration.
+    """
+
+    baseurl = "https://jsoc1.stanford.edu/data/aia/synoptic/"
+    known_wavelengths = [94, 131, 171, 193, 211, 304, 335, 1600, 1700, 4500]
+
+    required = {a.Time, a.Instrument, a.Level}
+    supported = required + {a.Sample, a.Wavelength, a.ExtentType}
+
+    @property
+    def info_url(self):
+        return self.baseurl
+
+    @classmethod
+    def register_values(cls):
+        adict = {
+            a.Instrument: [("AIA", "Data from the Atmospheric Imaging Assembly instrument.")],
+            a.Physobs: [
+                ("intensity", "Brightness or intensity of the solar atmosphere at different wavelengths.")
+            ],
+            a.Source: [("SDO", "The Solar Dynamics Observatory.")],
+            a.Wavelength: [(f"{wv:04d}", f"{wv} Å") for wv in cls.known_wavelengths],
+            a.Provider: [("JSOC", "Joint Science Operations Center at Stanford.")],
+            a.Level: [("synoptic", "Level 1.5 data processed for quicker analysis.")],
+            a.ExtentType: [("synoptic", "Level 1.5 data processed for quicker analysis.")],
+        }
+        return adict
+
+    def _can_handle_query(self, *query):
+        """
+        Determines whether this client can handle the given query.
+        """
+        required_attrs = {a.Instrument, a.Level}
+        supported_attrs = {a.Time, a.Instrument, a.Sample, a.Level, a.Wavelength, a.ExtentType}
+
+        # Check if all required attributes are present in the query
+        query_attrs = set(type(attr) for attr in query)
+        if not required_attrs.issubset(query_attrs):
+            return False
+
+        # Ensure there are no unsupported attributes in the query
+        if not query_attrs.issubset(supported_attrs):
+            return False
+
+        return True
+
+    def search(self, *args, **kwargs):
+        """
+        Perform a search for AIA synoptic data.
+        """
+        matchdict = self._get_match_dict(*args, **kwargs)
+        time_range = matchdict["time"]
+        start_time = time_range.start.datetime
+        end_time = time_range.end.datetime
+        sample = matchdict.get("sample", None)  # Get the sampling interval if provided
+        sample_minutes = (
+            sample.to(u.min).value if sample is not None else 2
+        )  # Default to 2-minute intervals if not provided
```

Comment:
Is there a need to default to 2 minutes if sample is not set?

Reply:
Only if we continue to use the construction of urls rather than scraping, I suppose. It's the maximum cadence the dataset contains.